src/projects/digital/api/circuit/src/internal/assembly/components/DemultiplexerAssembler.ts

```diff
@@ -0,0 +1,76 @@
+import {V, Vector} from "Vector";
+
+import {DigitalSim} from "digital/api/circuit/internal/sim/DigitalSim";
+import {AssemblerParams, AssemblyReason} from "shared/api/circuit/internal/assembly/Assembler";
+import {DigitalComponentConfigurationInfo} from "../../DigitalComponents";
+import {ComponentAssembler} from "shared/api/circuit/internal/assembly/ComponentAssembler";
+import {Schema} from "shared/api/circuit/schema";
+
+
+const MULTIPLEXER_HEIGHT_OFFSET = 0.5;
+export class DemultiplexerAssembler extends ComponentAssembler {
+    protected readonly sim: DigitalSim;
+
+    protected info: DigitalComponentConfigurationInfo;
+
+    public constructor(params: AssemblerParams, sim: DigitalSim) {
+        super(params, {
+            "inputs": (comp) => ({
+                origin: V(-this.getSize(comp).x/2, 0),
+                target: V(-this.getSize(comp).x/2 - this.options.defaultPortLength, 0),
+            }),
+            "outputs": (comp, index, total) => {
+                const x = this.getSize(comp).x/2
+                const y = -(index - ((total - 1) / 2)) / 2 + .25
+                return {
+                    origin: V(x, y),
+                    target: V(x + this.options.defaultPortLength, y),
+                }
+            },
+            "selects": (comp, index, total) => {
+                const { x: width, y: height } = this.getSize(comp)
+                const slope = MULTIPLEXER_HEIGHT_OFFSET / width;
+                const midPortOriginOffset = -height/2 + MULTIPLEXER_HEIGHT_OFFSET/2;
+                const x = (index - ((total - 1) / 2)) / 2;
+                const y = midPortOriginOffset - slope * x;
+                return {
+                    origin: V(x, y),
+                    target: V(x, -height/2 - this.options.defaultPortLength),
+                }
+            },
+        }, [
+            {
+                kind: "BaseShape",
+
+                dependencies: new Set([AssemblyReason.TransformChanged, AssemblyReason.PortsChanged]),
+                assemble:     (comp) => {
+                    const { x, y } = this.getSize(comp);
+                    const pos = this.getPos(comp);
+                    return {
+                        kind:   "Polygon",
+                        points: [
+                            V(x/2, y/2).add(pos),
```

Comment:
Will this work when rotating?

Reply:
No, and I'm now finding out the segments of the segment display don't rotate either. Is this something that should be handled here or in the Polygon renderer?